src/heartbeat/heartbeat.c

```diff
@@ -356,6 +423,12 @@ void send_hb_ping(mob_t* mob, uint8_t other_id, volatile bool* send_ping, volati
         send_hb_reset(other_id);
     }
 
+    if (*send_ping != false){
+        print("Error: send_ping is not false\n");
+    }
+    if (*received_resp != false){
+        print("Error: received_resp is not false\n");
+    }
     // Set flags to false just in case
     *send_ping = false;
     *received_resp = false;
```

Comment:
Below here in `run_hb()` (GitHub doesn't let me comment there), I think you need to check the flags for responding to restart info requests (unless I missed that happening somewhere else)